spec.bs

```diff
@@ -494,6 +506,218 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
+
+  : [=fenced frame config/fenced frame reporting metadata=]
+  :: null
+
+  : [=fenced frame config/exfiltration budget metadata=]
+  :: null
+
+  : [=fenced frame config/nested configs=]
+  :: a struct with the following members:
+    : [=nested configs/value=]
+    :: an empty [=list=] «»
+
+    : [=nested configs/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/embedder shared storage context=]
+  :: null
+
+</div>
+
+<div algorithm="fill in a pending fenced frame config">
+
+To <dfn>fill in a pending fenced frame config</dfn> given a [=fenced frame config=]
+|pendingConfig|, [=auction config=] |auctionConfig|, and [=leading bid info=] |winningBidInfo|:
+1. Let |winningBid| be |winningBidInfo|'s [=leading bid info/leading bid=].
+1. Set |pendingConfig|'s [=fenced frame config/mapped url=]'s [=mapped url/value=] to
+   |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
+1. Let |adSize| be |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/size=].
+1. If |adSize| is not null:
+   1. Set |pendingConfig|'s [=fenced frame config/content size=] to a struct with the following members:
+     : [=content size/value=]
+     :: |adSize| TODO: Resolve screen-relative sizes and macros and cast this properly.
+
+     : [=content size/visibility=]
+     :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/interest group descriptor=]'s
+   [=interest group descriptor/value=] to a struct with the following members:
+     : owner
+     :: |winningBid|'s [=generated bid/interest group=]'s [=interest group/owner=]
+
+     : name
+     :: |winningBid|'s [=generated bid/interest group=]'s [=interest group/name=]
+1. Set |pendingConfig|'s [=fenced frame config/fenced frame reporting metadata=] to a struct with
+   the following members:
+    : [=fenced frame reporting metadata/value=]
+    :: If |auctionConfig|'s [=auction config/component auctions=] is [=list/empty=] (i.e., if
+       there was no component auction), then a struct with the following members:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: true
+
+       Otherwise (i.e., if there was a component auction), a struct with the following members:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»,
+          "<code>component-seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: false
+
+    : [=fenced frame reporting metadata/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/on navigate callback=] to a closure of
+   [=asynchronously finish reporting=] partially evaluated with a reference to |pendingConfig|'s

```

Comment:
Regarding "partially evaluated," can you add a note below this step indicating when the other argument will be supplied and the algorithm will be called?

Reply:
I still see "partially evaluated". In any case, instead of using the language "closure" here can you just set the callback to a set of nested steps that make use of the variables it captures? That is a much more standard way of doing it in specs, at least until closures are fully specified https://github.com/whatwg/infra/issues/311.

Reply:
So you mean I would describe the nested steps in place here, rather than in a separate definition? I was trying to avoid that since the callback will potentially get a bit long, since there is a bunch of FLEDGE reporting stuff (not just fenced frame integration) that should go in there eventually.